usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -198,20 +200,27 @@ def handle(self, *args, **options):
             submission_ids = get_new_submission_ids(last_load_date)
             logger.info("Processing data for FABS starting from %s" % last_load_date)
 
-        if is_incremental_load and not submission_ids:
-            logger.info("No new submissions. Exiting.")
-
-        else:
+        if not is_incremental_load or submission_ids:
             with timer("obtaining delete records", logger.info):
                 ids_to_delete = get_fabs_records_to_delete(submission_ids, afa_ids, start_datetime, end_datetime)
 
             with timer("retrieving/diff-ing FABS Data", logger.info):
                 ids_to_upsert = get_fabs_transaction_ids(submission_ids, afa_ids, start_datetime, end_datetime)
 
-            update_award_ids = delete_fabs_transactions(ids_to_delete, do_not_log_deletions)
-            upsert_fabs_transactions(ids_to_upsert, update_award_ids)
+            externally_updated_award_ids = delete_fabs_transactions(ids_to_delete, do_not_log_deletions)
+            
+            if ids_to_upsert or externally_updated_award_ids:
+                update_award_ids = copy(externally_updated_award_ids)
```

Comment:
That call was as-is from before, and Tony's directive was to avoid too unneeded of changes, which makes me reluctant to change something just because it _won't_ make anything bad happen.

Reply:
I don't see how I can be blamed for this questionable code you are adding/modifying. :-)

Reply:
And while avoiding unnecessary edits is an admirable goal, this code was pilfered from another function.  It is yearning, nay, DEMANDING to be refactored in its new environment.

Once refactored, it will highlight another issue that I will bring up in a subsequent review if it is not discovered and fixed beforehand.